rholang/src/main/scala/coop/rchain/rholang/interpreter/Reduce.scala

```diff
@@ -175,19 +175,6 @@ object Reduce {
         evalImpl(term, newRand)
       }
 
-      def handle2(start : Int)(texpr : (Expr, Int)) : M[Unit] = {
```

Comment:
Ideally, all your commit messages would start with a capital letter, and there would be no commits undoing / refactoring code introduced in the same PR (refactors would be applied in-place, in the commit introducing the refactored code). If you can do a rebase dropping this and the previous commit, I'll appreciate one :) Otherwise that's just a general remark for the future (mostly the commit message part).

Reply:
Here the intention was to make small steps such that each of them are obviously correct. This is why I introduced this function first and then it was obvious that only the exprHandler was different compared to the other one so I parameterized the difference. I think these small steps are a very good way to help the reviewer understand the refactoring process.

Reply:
Yes, and another advantage of making those small steps is you don't even have to show things that don't make it to the final shape, like these two commits: one introducing handle2 and the other one removing it. How you avoid showing such steps - and achieve the *ultimate* ease for the reviewer (less material to process) - is by doing a rebase.

Unless you mean that showing the introduction and retraction itself would help me understand sth. All I understood from those two is you changed your mind in the process.